doc_stack/sphinx_manual_doc_layout

cspotcode opened the pull request:
Part of the PR stack described by #1797

cspotcode committed:
initial commit for doc_stack/sidebar

cspotcode committed:
initial commit for doc_stack/ux_polish

cspotcode committed:
initial commit for doc_stack/docstring_misc

cspotcode committed:
initial commit for doc_stack/sphinx_manual_doc_layout

cspotcode committed:
Change titles to be terser

cspotcode committed:
Fix repeated TOC entries

cspotcode committed:
Sneaking in unrelated fix: none of the other images are clickable

pushfoo commented:
Do we need to make any or changes to our config, contributing doc, or workflow after upgrading sphinx? The last version bump avoided 7.X because it had some sort of breaking changes (https://github.com/pythonarcade/arcade/pull/1745).

cspotcode commented:
> Do we need to make any or changes to our config, contributing doc, or workflow after upgrading sphinx? The last version bump avoided 7.X because it had some sort of breaking changes (#1745).

Not that I know if.  We can check if anything rendered incorrectly: https://api.arcade.academy/en/doc_structural_improvements/

cspotcode committed:
Collapse 3 "quick index" tables into one

cspotcode committed:
Add `from __future__ import annotations` everywhere

Command was isort arcade -a 'from __future__ import annotations'
But before running it, I find-and-replaced a # isort:skip comment onto every existing import
to prevent sorting

cspotcode committed:
Show class members in the order they appear in source code, include undocumented members (private are still omitted), and show links to classes' super-classes

cspotcode committed:
Always show grey highlight behind signatures, makes the doc sections easier to visually navigate

cspotcode committed:
Hide `ClassName.` prefix from class methods and attributes in TOC sidebar

cspotcode committed:
If only base class is `object`, don't add to docs; is just noise

cspotcode committed:
switch to grouped ordering of members; bysource was disorganized and it would be a lot of work to re-order our sources to a particular order

cspotcode committed:
Remove `property` prefix annotation from properties so that they appear as attributes in docs

cspotcode committed:
hierarchy re-org

cspotcode committed:
Remove extra headings from api pages

cspotcode committed:
Fix header underline

cspotcode committed:
Move docs into directories matching their new sidebar positions; move framebuffer doc back to tutorials do I don't have to feel guilty for moving it

cspotcode committed:
Fix repetition of "arcade is an easy-to-learn..." in PDF by wrapping the website's repetition in raw::html

cspotcode committed:
Fix doc warning about version.py being undocumented by adding logic to update_quick_index to exclude it explicitly

cspotcode committed:
fix

cspotcode committed:
Keep "API" in the title of "API Index" page, only shorten to "Index" in the html sidebar
